src/aiida_quantumespresso/parsers/xspectra.py

```diff
@@ -87,6 +87,28 @@ def parse(self, **kwargs):
         xy_data.set_y(y_arrays, y_names, y_units)
 
         parsed_data, logs = parse_stdout_xspectra(filecontent=out_file, codename='XSpectra')
+
+        # Parse some additional info which the stdout does not reliably report
+        input_params = self.node.get_incoming().get_node_by_label('parameters').get_dict()
```

Comment:
To get an input node, `node.inputs` is easier to read and more elegant.
```suggestion
        parameters = self.node.inputs.parameters.get_attribute('INPUT_XSPECTRA')
```
Since you also only ever use the `INPUT_XSPECTRA`, might as well get that subdictionary straight away.